lib/fit4ruby/Lap.rb

```diff
@@ -43,10 +54,27 @@ def initialize(records, previous_lap, field_values)
       set_field_values(field_values)
     end
 
-    def check(index)
+    def check(index, activity)
+      return if @num_length == 0
```

Comment:
This `return` should probably be moved down just before the `@first_length_index...` on line 70, I think.